src/plugins/data/common/search/search_source/search_source.ts

```diff
@@ -149,11 +150,22 @@ export class SearchSource {
    *****/
 
   /**
-   * internal, dont use
-   * @param searchStrategyId
+   * Used to make the search source overwrite the actual data view type for the
+   * specific requests done. This should only be needed very rarely, since it means
+   * e.g. we'd be treating a rollup index pattern as a regular one. Be very sure
+   * you understand the consequences of using this method before using it.
+   *
+   * @param overwriteType If `false` is passed in it will disable the overwrite, otherwise
+   *    the passed in value will be used as the data view type for this search source.
    */
-  setPreferredSearchStrategyId(searchStrategyId: string) {
-    this.searchStrategyId = searchStrategyId;
+  setOverwriteDataViewType(overwriteType: string | undefined | false) {
+    if (overwriteType === false) {
```

Comment:
wouldn't it be easier the other way around?

Reply:
Do you mean `if (overwriteType !== false) {`? I don't have a strong feeling in either direction, so happy to change it to whatever you feel reads better.

Reply:
I feel truthy-expressions-first reads better, but up to you